core/trino-main/src/main/java/io/trino/server/PluginManager.java

```diff
@@ -22,6 +22,7 @@
 import io.trino.security.AccessControlManager;
```

Comment:
>  (no implementation)

I think we need some implementation to justify the SPI changes. Can we get one? Is there anything that is commonly used?

Reply:
Hey @kokosing, not to my knowledge. As I stated initially, unlike basic-auth and x.509 there is no clear standard here. However, there are use-cases like Trino on a service mesh (ex. Istio) where it's tls terminated and so mTLS no longer becomes an authentication choice. The envoy proxy sidecars will convert the client certificate to an XFCC header so the only way to identify the client for an mTLS request then becomes the header. The actual authentication imo happens at the proxy level (trust client cert: yes -> convert to header, no -> sanitize header) and so, the logic that's in the Trino coordinator can be as rudimentary or sophisticated as you want (could use SPIRE from what I understand to check if the SPIFFE in the header is authorized) or just implicitly assume that if an XFCC header is present, the user is authenticated (the integrity of the header is maintained by proxy). This is why I'm against contributing our SPI implementation as it probably won't make sense for someone else. 

If not for this PR, is a header-based authentication method something on the Trino roadmap?